src/equations/shallow_water_exner_1d.jl

```diff
@@ -0,0 +1,618 @@
+    # Workaround to avoid division by zero, when computing the effective sediment height
+    if v_avg < eps()
+        h_s_avg = 0.0
+    else
+        h_s_avg = 0.5 * (q_s(u_ll, equations) / v_ll + q_s(u_rr, equations) / v_rr)
+    end
+
+    # Compute the eigenvalues using Cardano's formula
+    λ1, λ2, λ3 = eigvals_cardano(SVector(h_avg, h_avg * v_avg, h_s_avg), equations)
+
+    # Precompute some common expressions
+    c1 = g * (h_avg + h_s_avg)
+    c2 = g * (h_avg + h_s_avg / r)
+
+    # Eigenvector matrix
+    r31 = ((v_avg - λ1)^2 - c1) / c2
+    r32 = ((v_avg - λ2)^2 - c1) / c2
+    r33 = ((v_avg - λ3)^2 - c1) / c2
+    R = @SMatrix [[1 1 1]; [λ1 λ2 λ3]; [r31 r32 r33]]
+
+    # Inverse eigenvector matrix
+    d1 = (λ1 - λ2) * (λ1 - λ3)
+    d2 = (λ2 - λ1) * (λ2 - λ3)
+    d3 = (λ3 - λ2) * (λ3 - λ1)
+    R_inv = @SMatrix [(c1 - v_avg^2 + λ2 * λ3)/d1 (2 * v_avg - λ2 - λ3)/d1 c2/d1;
+                      (c1 - v_avg^2 + λ1 * λ3)/d2 (2 * v_avg - λ1 - λ3)/d2 c2/d2;
+                      (c1 - v_avg^2 + λ1 * λ2)/d3 (2 * v_avg - λ2 - λ1)/d3 c2/d3]
+
+    # Eigenvalue absolute value matrix
+    Λ_abs = @SMatrix [abs(λ1) z z; z abs(λ2) z; z z abs(λ3)]
+
+    # Compute dissipation
+    diss = SVector(-0.5 * R * Λ_abs * R_inv * (u_rr - u_ll))
+
+    return SVector(diss[1], diss[2], diss[3])
+end
+
+@inline function Trixi.max_abs_speed_naive(u_ll, u_rr, orientation::Integer,
+                                           equations::ShallowWaterExnerEquations1D)
+    return max(eigvals_cardano(u_rr, equations)..., eigvals_cardano(u_ll, equations)...)
+end
+
+@inline function Trixi.max_abs_speeds(u, equations::ShallowWaterExnerEquations1D)
+    return maximum(eigvals_cardano(u, equations))
+end
+
+#Helper function to extract the velocity vector from the conservative variables
+@inline function Trixi.velocity(u, equations::ShallowWaterExnerEquations1D)
+    h, hv, _ = u
+
+    v = hv / h
+
+    return v
+end
+
+# Compute the sediment discharge for Shields stress models
+@inline function q_s(u,
+                     equations::ShallowWaterExnerEquations1D{T, S,
+                                                             ShieldsStressModel{T}}) where {
+                                                                                            T,
+                                                                                            S
+                                                                                            }
+    (; gravity, rho_f, rho_s, porosity_inv) = equations
+    (; m_1, m_2, m_3, k_1, k_2, k_3, theta_c, d_s) = equations.sediment_model
+
+    theta = rho_f * abs(shear_stress(u, equations)) / (gravity * (rho_s - rho_f) * d_s)  # Shields stress
+
+    Q = d_s * sqrt(gravity * (rho_s / rho_f - 1.0) * d_s) # Characteristic discharge
+
+    return porosity_inv * Q * sign(theta) * k_1 * theta^m_1 *
+           (max(theta - k_2 * theta_c, 0.0))^m_2 *
+           (max(sqrt(theta) - k_3 * sqrt(theta_c), 0.0))^m_3
```

Comment:
```suggestion
    return (porosity_inv * Q * sign(theta) * k_1 * theta^m_1 *
           (max(theta - k_2 * theta_c, 0.0))^m_2 *
           (max(sqrt(theta) - k_3 * sqrt(theta_c), 0.0))^m_3)
```
Add some parentheses for safety.